course-v2/assets/js/mobile_course_drawers.ts

```diff
@@ -0,0 +1,48 @@
+enum SWIPE_DIRECTION {
+  Left = "L",
+  Right = "R"
+}
+
+export const initCourseDrawersClosingViaSwiping = () => {
+  elementSwiping(
+    "mobile-course-nav",
+    "mobile-course-nav-toggle",
+    SWIPE_DIRECTION.Left
+  )
+  elementSwiping(
+    "course-info-drawer",
+    "mobile-course-info-toggle",
+    SWIPE_DIRECTION.Right
+  )
+}
+
+/**
+ * It adds touch event listener to the element and clicks the button when swiped in the mentioned direction.
+ *
+ * @param {string} elementId element on which touch eventlistenter is added.
+ * @param {string} buttonId This button will be clicked on swiping
+ * @param {string} swipeDirection L= Swipe left, R= Swipe Right
+ */
+const elementSwiping = (
+  elementId: string,
+  buttonId: string,
+  swipeDirection: SWIPE_DIRECTION
+) => {
+  const element = document.getElementById(elementId)
+  let touchstartX = 0
+  let touchendX = 0
+
+  element?.addEventListener("touchstart", e => {
```

Comment:
Rather than using optional chaining, I usually throw an error in situations like this if `element` is null.

Rationale: `element` should not be null. If it is, that's an error. We can either throw the error and get a better message and typing below, or we exit the function / use optional chaining. Then there's no visible error, but the site still behaves wrong.